src/systems/joint_position_controller/JointPositionController.cc

```diff
@@ -271,14 +273,47 @@ void JointPositionController::PreUpdate(
     bool warned{false};
     for (const std::string &name : this->dataPtr->jointNames)
     {
-      Entity joint = this->dataPtr->model.JointByName(_ecm, name);
+      // First try to resolve by scoped name.
+      Entity joint = kNullEntity;
+      auto entities = entitiesFromScopedName(
+          name, _ecm, this->dataPtr->model.Entity());
+
+      if (!entities.empty())
+      {
+        if (entities.size() > 1)
+        {
+          gzwarn << "Multiple joint entities with name ["
+                << name << "] found. "
+                << "Using the first one.\n";
+        }
+        joint = *entities.begin();
+
+        // Validate
+        if (!_ecm.EntityHasComponentType(joint, components::Joint::typeId))
+        {
+          gzerr << "Entity with name[" << name
+                << "] is not a joint\n";
+          joint = kNullEntity;
+        }
+        else
+        {
+          gzdbg << "Identified joint [" << name
+                << "] as Entity [" << joint << "]\n";
+        }
+      }
+      else
+      {
+        // Otherwise attempt to resolve by unscoped name
+        joint = this->dataPtr->model.JointByName(_ecm, name);
```

Comment:
Is there a reason for keeping this? If `entitiesFromScopedName` doesn't find the joint, I don't think this will.

Reply:
Thanks for the review. You're right - removed as redundant in 0a4fb74b0. I've rebased on gz-sim7_7.3.0 squashed and force pushed to tidy things up.

Reply:
Ok - handy to know. I'll use the button to merge if needed next time. It's a habit from ArduPilot where they like to have a linear history in main and their vehicle branches, which I think helps them when using git bisect if they have a break in the flight controller code.

Reply:
Yeah, I think we like linear history too (except for forward ports from earlier versions), but we achieve that by squashing all the commits in a PR into a single commit when it's time to merge the PR.